Formula/hisat2.rb

```diff
@@ -2,9 +2,12 @@ class Hisat2 < Formula
   # cite Kim_2015: "https://doi.org/10.1038/nmeth.3317"
   desc "Graph-based alignment to a population of genomes"
   homepage "https://ccb.jhu.edu/software/hisat2/"
-  url "ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/downloads/hisat2-2.1.0-source.zip"
+  url "https://github.com/infphilo/hisat2/archive/0f01dc6397a.zip"
   sha256 "89a276eed1fc07414b1601947bc9466bdeb50e8f148ad42074186fe39a1ee781"
 
+  fails_with :clang
+  depends_on "gcc" if OS.mac? # compilation with clang fails
```

Comment:
If you have access to macOS, check `brew linkage hisat2` to confirm that it does in fact link to `gcc` libraries. If there's no linkage, the dependency is not required.